### ISSUE TITLE
`TestCompletenessReset` flakiness

### ISSUE THREAD

**Issue by greedy52** [gravitational/teleport#19495]
#### Link(s) to logs

https://console.cloud.google.com/cloud-build/builds/be090b07-b74f-4884-a3ba-3913b3f3f655?project=ci-account

#### Relevant snippet

```
=== CONT  TestCompletenessReset
    testing.go:1319: race detected during execution of test
```

**Comment by zmb3** [comment]
This should be fixed by #19390

**Comment by greedy52** [comment]
Duplicate of #19294
